api/src/components/authorization/controller.ts

```diff
@@ -65,7 +65,7 @@ export const authorize = async (event: I.APIRequest<I.AuthorizeRequestBody>): Pr
         // eslint-disable-next-line @typescript-eslint/dot-notation
         const works = userInformation['activities-summary'].works['group'].map((work) => ({
             title: work['work-summary'][0].title.title.value || null,
-            doi:
+            doi: work['work-summary'][0]['external-ids'] === null ? null :
```

Comment:
There's no need to check for null, you can use optional chaining for all the nested properties and put a "|| null" at the end.

From what I remember, we agreed to use optional chaining for all these mappings until we refine the login flow, because there's more stuff that can break. Any thoughts @karmseveeratjisc ??

Reply:
ideally yes

Reply:
I have updated to use optional chaining for that particular mapping, what is meant by all these mappings ?

Reply:
all the other places in that same file we do similar (accessing nested properties) , even for that one we could go mad and put something like

work['work-summary']?.[0]?.['external-ids']?.['external-id]

I think, as according to the spec external-ids is also optional

e.g. line 67 could be

work['work-summary']?.[0]?.title?.title?.value || null,